apps/homescreen/js/app.js

```diff
@@ -435,6 +453,21 @@ const SETTINGS_VERSION = 0;
         handleRoleChange(icon.app, container);
       } else {
         icon.bookmark = appOrBookmark;
+
+        IconsHelper.getIconBlob(icon.bookmark.url, this.iconSize,
```

Comment:
Previously in this patch the logic was into the component itself, but due to dependencies, I had to move it into Gaia tree. A discussion with @albertopq convinced me to do it that way instead.
At the moment the only icon image that the `gaia-site-icon` element knows about is the one in `bookmark.icon`. It is not optimised for the space available on the homescreen.